src/testhost.x86/DefaultEngineInvoker.cs

```diff
@@ -32,6 +32,8 @@ internal class DefaultEngineInvoker :
         /// </summary>
         private const int ClientListenTimeOut = Timeout.Infinite;
```

Comment:
where is ClientListenTimeout used ? And can it be removed ?

Reply:
ClientListenTimeout is used by testhost to connect to vstest.console. We are waiting infinitely here so as to allow testhost to keep on polling vstest for comm. However vstest.console only waits for 60secs as default(this is configurable) to connect to testhost. If this does not connect to testhost in this time, it will itself kill testhost process. So this timeout will not cause deadlock

Reply:
ClientListenTimeout is used by testhost to connect to vstest.console. We are waiting infinitely here so as to allow testhost to keep on polling vstest for comm. However vstest.console only waits for 60secs as default(this is configurable) to connect to testhost. If this does not connect to testhost in this time, it will itself kill testhost process. So this timeout will not cause deadlock

Reply:
Let us not have infinite timeouts anywhere in the code.. keep the timeout then the twice of the vstest.console.exe timeout.. but not infinite.. 

---
In reply to: [165572248](https://github.com/Microsoft/vstest/pull/1406#discussion_r165572248) [](ancestors = 165572248)

Reply:
Like I said vstest.console will kill testhost process if cannot connect to it. So keeping an infinite timeout here is not harmful. Also the timeout at vstest.console is configurable, & not here. What I can do however is make this also configurable using same environment variable as we use for vstest.console, but then again allowing testhost to wait indefinitely to connect to vstest.console is not harfmful.

Let me know if I still need to make this change